bin/si-sdf-api-test/tests/1-create_and_apply_across_change_sets.ts

```diff
@@ -1,75 +1,96 @@
 import assert from "node:assert";
 import { SdfApiClient } from "../sdf_api_client.ts";
-import { extractSchemaVariant, getSchemaVariants } from "../test_helpers.ts";
-import abandon_all_change_sets from "../benchmark/abandon_all_change_sets.ts";
+import { abandon_all_change_sets, createComponent, createComponentPayload, eventualMVAssert, getVariants, getViews, installModule } from "../test_helpers.ts";
 
 export default async function create_and_and_apply_across_change_sets(
   sdfApiClient: SdfApiClient,
 ) {
   return await create_and_and_apply_across_change_sets_inner(sdfApiClient);
 }
-
+const TEST_RESOURCE_ACTION_MODULE_ID = "01JMJB1QW3M8D7MNN8NGBR739J";
 async function create_and_and_apply_across_change_sets_inner(
   sdf: SdfApiClient,
 ) {
   sdf.listenForDVUs();
-  console.log("cleaning up any existing components...");
+  console.log("cleaning up any existing components and open change sets...");
   await cleanupHead(sdf);
   console.log("creating changesets with the test component in them...");
   let changeSetIds: string[] = [];
   let err: Error | undefined = undefined;
+
   try {
+    // Create 10 change sets, each with one test component in them
     for (let i = 0; i < 10; i++) {
       const changeSetId = await createChangeSet(sdf);
       changeSetIds.push(changeSetId);
+      const views = await getViews(sdf, changeSetId);
+      const defaultView = views.find((v: any) => v.isDefault);
+      assert(defaultView, "Expected to find a default view");
+
+
+      let schemaVariants = await getVariants(sdf, changeSetId);
+
+      // first let's make sure our test asset is installed
+      // this is really just so we can run this against any workspace
+      if (!schemaVariants.installed.some((v: any) => v.schemaName === "TestResourceActions")) {
+        const installResult = await installModule(sdf, changeSetId, TEST_RESOURCE_ACTION_MODULE_ID);
+        console.log("Install Result:", installResult);
+        assert(installResult[0]?.schemaVariantId, "Expected to get a schemaVariantId after installing TestResourceActions module");
+        const installedVariantId = installResult[0].schemaVariantId;
+        console.log("Installing TestResourceActions schema variant...", installResult);
+        await eventualMVAssert(
+          sdf,
+          changeSetId, "SchemaVariantCategories",
+          sdf.workspaceId,
+          (mv) => {
+            const installedList: string[] = mv.categories.flatMap((c: any) => c.schemaVariants.filter((v: any) => v.type === "installed").map((v: any) => {
+              return v.id;
+            }));
+            console.log("Installed Schema Variants:", installedList);
+            console.log("Expected Schema Variant ID:", installedVariantId);
+            return installedList.some((v: any) => v === installedVariantId);
+          },
+          "Expected TestResourceActions schema variant to be installed",
+        );
+        schemaVariants = await getVariants(sdf, changeSetId);
```

Comment:
All 10 change set calls will be updating this. Is that ok?

Reply:
The iteration creates and applies 10 change sets and from what I am reading, we re-assign the variants on each iteration.